### ISSUE TITLE
Gallery+ links in Gallery

### ISSUE THREAD

**Issue by tessus** [owncloud/gallery#345]
@jancborchardt suggested I should ask this question here.

I've noticed that Gallery+ will be integrated into oc 8.2.0 and I'm curious what will happen to all the links that were created by Gallery+.

e.g. right now I have a link `https://<server>/index.php/apps/galleryplus/s/Y3hngafqvdLBR7g`
Will this link still work in 8.2.0?
##

**Comment by oparoz** [comment]
Luckily, Éric Seigne has contacted me to tell me about this issue.

There are 3 solutions:
1. Use a rewrite Rule: [Migrating from Gallery and vice versa](https://github.com/owncloud/gallery/wiki/Migrating-from-Gallery--and-vice-versa)
2. Keep using Gallery+, which will also have a version for 8.2, but with more regular updates
3. Use an app which installs as galleryplus, but only does redirects

**Comment by tessus** [comment]
Hi Oliver. 

Thanks for the info. Although item 2 keeps me a bit puzzleld. I assumed that integrating Gallery+ back into the product would also replace the Gallery code and make Gallery+ the one and only code base. But the fact that Gallery+ will still be available as a separate product contradicts my assumption.
This is rather confusing.
Please make sure that differences between Gallery+ (in oc 8.2.0) and Gallery+ (separate product) are mentioned in the documentation.

**Comment by oparoz** [comment]
There will be announcement on blogs and in the forums, explaining what users can do, depending on their preferences.

You have two options: 
- Stick with the default Gallery and get new features every 3 months or so. Per example. If I implement something new next week, it will be in 9.0, but most users don't trust the dot zero releases, so will upgrade when it reaches dot one or dot two, so that means a feature released in September 2015 will only be available for some users between January and March 2016. So, almost 6 months later for cautious people. Granted, those users care more about stability than new features.
- Switch to Gallery+ (different repository), which uses the same code base, but which should get new releases monthly. I got sidetracked because of the integration into the official distribution and the conference. Also there are not enough people testing master, so it takes me a lot of time to test a lot of scenarios. But I plan on communicating more about pre-releases to make sure people start using the new features early and I will automate the list of acceptance tests I run before a release. Of course, new features might contain more bugs, but the monthly releases should ensure that you don't have to wait too long to get fixes.

Gallery+ might also get some different features or UI elements over time, depending on feedback and my needs, but that's not a priority atm.

I hope this help clarify the situation.

**Comment by tessus** [comment]
Thanks again.

2 last (maybe dumb) questions:
- since Gallery and Gallery+ cannot be run in parallel why not use the same external links for the app (since it is the same code base anyway)?
- the oc Apps store uses the name Gallery (v13.0.0.0) for Gallery+ and Pictures for Gallery. How do we distinguish the 2 apps in the future?

**Comment by oparoz** [comment]
> since Gallery and Gallery+ cannot be run in parallel why not use the same external links for the app (since it is the same code base anyway)?

I'll see if I can add Gallery routes to Gallery+

> the oc Apps store uses the name Gallery (v13.0.0.0) for Gallery+ and Pictures for Gallery. How do we distinguish the 2 apps in the future?

It will be called Gallery+ again.

**Comment by tessus** [comment]
> I'll see if I can add Gallery routes to Gallery+

Great. Since they are basically the same (except that Gallery+ might have additional features) this would really make sense. In this case people can even switch back and forth w/o having to worry about their existing links.

Thank you very much for your help.

**Comment by deMattin** [comment]
It's actual really a bit confusing.
I have a fresh oc 8.1.3 on debian 8 and there it looks like the following:
you can decide, if you want to
1. Install Pictures (old Gallery) which presenting an upgrade button to Version 13 (not named but is indeed gallery+ here!)
2. Install Gallery+ Version 13 instead of Pictures

If you Install Gallery+, the Pictures app is not selectable any more (for security reason, not to install both together, which doesn't work).

But you have both folders in apps folder `/gallerplus` and `/gallery`

Questions that arise to someone, if checking folders:
Is the folder `gallery` the not used pictures app folder? Why isn't it named `pictures`?
Or is the pictures folder not there, because pictures is not installed?
Is the gallery+ folder the actual used folder for the app called `gallery`, if you update pictures to ver 13 or is it galleryplus folder when you use the owncloud app selected galleryplus and gallery folder is updated to ver 13 (being galleryplus in gallery folder)?

So all in all you use an app called gallery in owncloud useage that has a folder gallerplus
and another app called pictures, using the folder gallery.
And "pictures app update to ver 13" is the same as deinstalling pictures and installing galleryplus (but with a remaining folder gallery with old pictures app)

I crashed my productive owncloud instance a month ago by messing a git install of galleryplus and clicking on galleryplus update in app selector in owncloud admin. And my tries to repair corrupted all because owncloud didn't know any more, which app was active (showing me a physical not existing galler+ as active!)

I _think_ (but am not sure), that I understand now the relations but noone can get this, if not knowing the development of gallery+ ... ;)

But one question is remaining for me:
How am I able to install additionally to an official gallery (named gallery in Owncloud, is the same as `pictures update` or  galleryplus in app selector and is using folder galleryplus on default!) next to a git gallery+ from interfasys or owncloud master?
This should be possible, because testing in a productive environment is much better than testing on isolated test installs.

**Comment by oparoz** [comment]
> I'll see if I can add Gallery routes to Gallery+

Not possible AFAIK. The router checks if the app is enabled and won't load the route. @LukasReschke, please confirm :) (Trying to route `apps/gallery/s/{token}` to `galleryplus.page_controller.public_index`)

**Comment by oparoz** [comment]
>  Install Pictures (old Gallery) which presenting an upgrade button to Version 13 (not named but is indeed gallery+ here!)

That's really confusing. I'll try with the VM to see if I can confirm that behaviour. Could be an app store issue or could be that you have the Gallery+ ocsid associated with the gallery namespace for some reason

> If you Install Gallery+, the Pictures app is not selectable any more (for security reason, not to install both together, which doesn't work).

Could be related to the oscid problem described above.

> Is the folder gallery the not used pictures app folder? Why isn't it named pictures?

No idea why this name was picked for Pictures.

> So all in all you use an app called gallery in owncloud useage that has a folder gallerplus

Yes, that may have been a mistake. 

> How am I able to install additionally to an official gallery next to a git gallery+ from interfasys or owncloud master?

I'd recommend keeping /apps read-only and to create another folder for apps you install.
Normally, on 8.1, you should not be able to upgrade Pictures to Gallery+. You get new versions of Pictures with new releases of ownCloud and you get new versions of Gallery+ through the app store.
On 8.2, you can have the same setup and you'll get Gallery instead of Pictures in your gallery folder. Gallery+ will stay in the galleryplus folder.

**Comment by deMattin** [comment]
Maybe something messed with ocsid in the case, where I corrupted my productive installation.
But on my fresh Debian 8 installation the above described behaviour is default.

So I will wait for 8.2 to go further with "experiments" and will check, how to use different pathes than `/apps` for some apps and also will investigate ocsid.
The next 4 weeks I'm on the road and so better no dangerous experiments ... ;)

**Comment by oparoz** [comment]
![pic_update_is_wrong](https://cloud.githubusercontent.com/assets/323220/10070483/1ba5fe34-62b3-11e5-8ad8-1e84ce0756c0.png)

Really confusing... I've updated the info.xml of this app, so that it doesn't show up in older installs

**Comment by deMattin** [comment]
Yes, this is what I've been talking about ...

**Comment by oparoz** [comment]
This issue was moved to nextcloud/gallery#79
